File: tests/src/main/java/com/github/klee0kai/test/gc/di/ann/ContextGcScope.java
package com.github.klee0kai.test.gc.di.ann;


import javax.inject.Scope;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Scope
@Retention(value = RetentionPolicy.CLASS)
@Target(value = ElementType.METHOD)
public @interface ContextGcScope {
}

File: test_kotlin/src/main/kotlin/com/github/klee0kai/test/gc/di/ann/ContextGcScope.kt
package com.github.klee0kai.test.gc.di.ann

import javax.inject.Scope

@Scope
@Retention(AnnotationRetention.BINARY)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class ContextGcScope 